crates/lox_core/src/bodies/cip/xy06.rs

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2023. Helge Eichhorn and the LOX contributors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+mod amplitudes;
+mod luni_solar;
+mod planetary;
+mod polynomial;
+
+use crate::bodies::fundamental::iers03::{
+    general_accum_precession_in_longitude_iers03, mean_moon_sun_elongation_iers03,
+};
+use crate::bodies::{Earth, Jupiter, Mars, Mercury, Moon, Neptune, Saturn, Sun, Uranus, Venus};
+use crate::math::arcsec_to_rad;
+use crate::time::intervals::TDBJulianCenturiesSinceJ2000;
+use crate::types::Radians;
+
+/// A convenient type for performing batch mathematical operations on X and Y components. This
+/// type may change or become unexported as the needs of upstream components become clearer.
+pub type XY = [f64; 2];
+
+const MAX_POWER_OF_T: usize = 5;
+
+type PowersOfT = [f64; MAX_POWER_OF_T + 1];
+
+type FundamentalArgs = [Radians; 14];
+
+type MicroArcsecond = f64;
+
+#[derive(Default)]
+struct NutationComponents {
+    planetary: XY,
+    luni_solar: XY,
+}
+
+/// (X, Y) coordinates of the Celestial Intermediate Pole (CIP) using the the IAU 2006 precession
+/// and IAU 2000A nutation models.
+pub fn xy(t: TDBJulianCenturiesSinceJ2000) -> XY {
+    let powers_of_t = powers_of_t(t);
+    let fundamental_args = fundamental_args(t);
+    let polynomial_components = polynomial_components(&powers_of_t);
+    let nutation_components = nutation_components(&powers_of_t, &fundamental_args);
+    calculate_cip_unit_vector(&polynomial_components, &nutation_components)
+}
+
+fn powers_of_t(t: TDBJulianCenturiesSinceJ2000) -> PowersOfT {
+    let mut tn: f64 = 1.0;
+    let mut powers_of_t = [0.0; MAX_POWER_OF_T + 1];
+    for pow in powers_of_t.iter_mut() {
+        *pow = tn;
+        tn *= t;
+    }
+    powers_of_t
+}
+
+/// The output of the CIP calculation is dependent on the ordering of these arguments. DO NOT EDIT.
```

Comment:
Does it make sense to move this comment right next to the value? One might not notice that the comment relates to it.

Is there a test which fails if the order gets changed?